src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckClassVisitor.kt

```diff
@@ -121,13 +122,159 @@ internal class LincheckClassVisitor(
+                    PUTSTATIC -> if (state == State.LOADED_HITS_ON_STACK_STATIC) {
+                        state = State.INITIAL
+                        initialAdapter.visitFieldInsn(opcode, owner, name, desc)
+                        return
+                    }
+                }
+            }
+            super.visitFieldInsn(opcode, owner, name, desc)
+        }
+
+        override fun visitJumpInsn(opcode: Int, label: Label?) {
+            when (opcode) {
+                IFNONNULL -> if (state == State.LOADED_HITS_ON_STACK_STATIC) {
+                    state = State.INITIAL
+                    initialAdapter.visitJumpInsn(opcode, label)
+                    return
+                }
+            }
+            super.visitJumpInsn(opcode, label)
+        }
+
+        override fun visitInsn(opcode: Int) {
+            when (opcode) {
+                BASTORE, IASTORE -> if (state == State.PREPARE_ASSIGN_TO_HITS) {
+                    state = State.LOADED_HITS_IN_LOCALS
+                    initialAdapter.visitInsn(opcode)
+                    return
+                }
+                IALOAD -> if (state == State.PREPARE_ASSIGN_TO_HITS) {
+                    initialAdapter.visitInsn(opcode)
+                    return
+                }
+            }
+
+            super.visitInsn(opcode)
+        }
+
+        override fun visitVarInsn(opcode: Int, index: Int) {
+            when (opcode) {
+                ASTORE -> {
+                    if (state == State.LOADED_HITS_ON_STACK || state == State.LOADED_HITS_ON_STACK_STATIC) {
+                        state = State.LOADED_HITS_IN_LOCALS
+                        localVariableIndex = index
+                    }
+                }
+                ALOAD -> if (state == State.LOADED_HITS_IN_LOCALS && localVariableIndex == index) {
+                    state = State.PREPARE_ASSIGN_TO_HITS
+                }
+            }
+
+            super.visitVarInsn(opcode, index)
+        }
+
+        override fun visitMethodInsn(
+            opcode: Int,
+            owner: String,
+            name: String,
+            descriptor: String,
+            isInterface: Boolean
+        ) {
+            val methodSignature = "$owner.$name$descriptor"
+            when (opcode) {
+                INVOKESTATIC -> if (
+                    state == State.INITIAL &&
+                    methodSignature == COVERAGE_GET_HITS_MASK_METHOD_SIGNATURE
+                ) {
+                    state = State.LOADED_HITS_ON_STACK_STATIC
+                    initialAdapter.visitMethodInsn(opcode, owner, name, descriptor, isInterface)
+                    return
+                }
+            }
+
+            super.visitMethodInsn(opcode, owner, name, descriptor, isInterface)
+        }
+
+        override fun visitEnd() {
+            state = State.INITIAL
+            localVariableIndex = -1
+
+            super.visitEnd()
+        }
+    }
+
+    private class CoroutineCancellabilitySupportMethodTransformer(
```

Comment:
Is this related to this PR?

Reply:
`CoroutineCancellabilitySupportMethodTransformer` was moved into `transformers/CoroutineSupportTransformers.kt` file, please remove it from here.